build/pip/setup.py

```diff
@@ -0,0 +1,122 @@
+"""The setup installs Gauge CLI through pip."""
+from setuptools import setup
+from os import path
+from setuptools.command.install import install
+import json
+import platform
+import glob
+import zipfile
+import shutil
+import os
+import sys
+try:
+    import requests
+
+except ImportError:
+    os.system('python -m ensurepip --default-pip')
+    os.system('python -m pip install requests')
+    import requests
+
+# TODO: bumped manually for now, wish to be automatically picked from source
+release_ver = '1.0.2'
+base_url = 'https://api.github.com/repos/getgauge/gauge/releases'
+
+# Code to determine Architecture and OS where Gauge is being installed
+arch_map = {"ia32": "x86", "x64": "x86_64"}
+os_map = {"Darwin": "darwin", "linux": "linux", "win32": "windows"}
+if '64' in platform.architecture()[0]:
+    arch = arch_map['x64']
+else:
+    arch = arch_map['ia32']
+os_name = os_map[platform.system()]
+if os_name is 'win32':
+    gauge_file = 'gauge.exe'
+else:
+    gauge_file = 'gauge'
+
+latest_version = json.loads(requests.get(base_url).text)[0]['tag_name']
+
+
+class CustomInstallCommand(install):
+    """Customized setuptools install command to download and setup."""
+
+    def _gauge_package_fetch(self):
+        package_name = 'gauge-%s-%s.%s' % (release_ver, os_name, arch)
+        package_url = base_url.replace('api.', '').replace('/repos', '') + '/download/v%s/%s.zip' % (release_ver, package_name)
+        r2 = requests.get(package_url)
+        with open("gauge.zip", "wb") as download:
+            download.write(r2.content)
+
+    def gauge_main_to_path(self):
+        """Place Gauge CLI into Scripts Directory."""
+        file_list = glob.glob('*')
+        if 'bin' in file_list:
+            shutil.rmtree('bin')
+        self._gauge_package_fetch()
+        with zipfile.ZipFile('gauge.zip', "r") as z:
+            z.extractall()
+        target_path = os.path.join(self.install_scripts, gauge_file)
+        source_path = os.path.join(os.getcwd(), gauge_file)
+        if os.path.isfile(target_path):
+            os.remove(target_path)
+        if os.path.isfile(source_path):
+            self.move_file(source_path, target_path)
+
+            if os_name is not "win32":
+                if sys.version_info[0] == 3:
+                    os.chmod(target_path, 0o755)
+                if sys.version_info[0] < 3:
+                    os.chmod(target_path, 755)
+
+    def run(self):
+        """Custom Install / Run Commands."""
+        self.gauge_main_to_path()
+        install.run(self)
+
+
+this_directory = path.abspath(path.dirname(__file__))
+long_description = None
```

Comment:
PLease refer https://python-packaging.readthedocs.io/en/latest/metadata.html#a-readme-long-description for long descriptions.
Use `manifest.in` instead of reading the file (it will avoid the try/except).

Reply:
done, updated the code